core/src/Cuda/Kokkos_Cuda_KernelLaunch.hpp

```diff
@@ -398,9 +398,10 @@ struct CudaParallelLaunchKernelInvoker<
       params.blockDim       = block;
       params.gridDim        = grid;
       params.sharedMemBytes = shmem;
-      params.func           = (void*)base_t::get_kernel_func();
-      params.kernelParams   = (void**)args;
-      params.extra          = nullptr;
+      // Casting a function pointer to a data pointer...
+      params.func         = reinterpret_cast<void*>(base_t::get_kernel_func());
+      params.kernelParams = const_cast<void**>(args);
```

Comment:
can cuda modify this? I assume not but this will be bad UB if it can

Reply:
According the [documentation](https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__GRAPH.html), they copy the data.